netlify-cms.config.yml

```
@@ -95,3 +95,89 @@ collections:
     label: Past
 
   preview_path: api/preview?page=about&ref=cms/team/{{slug}}
+
+
+- name: history
+
+  folder: history
+  extension: md
+  format: yaml-frontmatter
+
+  media_folder: '/images/history'
+
+  create: true
+  delete: true
+
+  label: 🕰 History
+  label_singular: event in history
+  summary: '{{date}} - {{title}}'
+
+  fields:
+  - name: date
+    label: Date
+    widget: datetime
+    format: YYYY-MM-DD
+    date_format: YYYY-MM-DD
+    time_format: ''
+    default: ''
+    hint: >
+      All events on the page are sorted in chronologic order.
+      The day is not displayed, therefore it doesn't have to be correct.
+    picker_utc: true
+  - name: title
+    label: Event
+    widget: string
+    hint: >
+      The title for the event
+  - name: body
+    widget: markdown
+    buttons: [bold, italic, link, quote, bulleted-list, numbered-list]
+    label: Description
+    hint: >
+      Try to keep this concise, it should be approximately 2-3 lines on the website.
+  - name: image
+    label: Image
+    widget: object
+    summary: 'Image: {{alt}}'
+    hint: >
+      A good illustration of what the event is about.
+      Despite it being optional, having illustrations is preferable.
+    fields:
+      - name: src
+        label: Source
+        widget: image
+        allow_multiple: false
+        required: false
+        hint: >
+          Provide a link to the image or upload as a file.
+      - name: alt
+        label: Description
+        widget: string
+        required: false
+        hint: >
+          A short description. Necessary in case the user can't see the image.
```

```suggestion
      - name: alt
        widget: string
        required: false
        label: Alternative text
        hint: >
          Description of what is actually displayed on the image.
          It helps tech-assistive users who can't see the screen understand
          what is on the picture.
```

"Alternative text" is more common, I'd say. Also, HTML requires it. Images are not going to be decorative, so `alt` must not be empty.

Is there a way to make whole object optional but fields inside all required?

---

Not possible to my knowledge. [Here is the relevant issue](https://github.com/netlify/netlify-cms/issues/2103).